### PR TITLE
ASM Customer List Facade refactor

### PR DESCRIPTION
**Pull request by niehuayang**
JIRA: https://jira.tools.sap/browse/CXSPA-2409

### CODE REVIEW

File: feature-libs/asm/core/connectors/asm.adapter.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { Observable } from 'rxjs';
 import {
   CustomerSearchOptions,
   CustomerSearchPage,
-} from '../models/asm.models';
+} from '../../root/model/asm.models';
```
Comment (ercultimate) [inline feature-libs/asm/core/connectors/asm.adapter.ts]: This import is invalid.  It should be `@spartacus/asm/root`.  There are many other instances of this.  I'd recommend running a search for "../root/model" in the asm folder.

This file (`asm.adapter.ts`) is part of the `@spartacus/asm/core` module.  Relative imports will only work for files that are part of the same module.  The `asm.models.ts` file was moved to the `@spartacus/asm/root` module, so we need to use a fully qualified import.  The auto-import is not smart enough to recognize this and gave you the same bad import for all the files.

This issue would only appear once you tried to compile the Spartacus project into the Angular library using `npm build:libs`.

**Review by ercultimate — CHANGES_REQUESTED** [review]
This is looking pretty good.  Just the imports to fix and then documenting the API changes in the migration document.

File: feature-libs/asm/core/connectors/asm.adapter.ts (hunk 1 above)
Comment (niehuayang) [reply feature-libs/asm/core/connectors/asm.adapter.ts]: Ah, I see, thanks very much for providing more details, and now I have corrected it.

File: feature-libs/asm/core/facade/asm.service.ts [hunk 2]
```diff
@@ -53,31 +53,6 @@ export class AsmService {
...
-   * Returns the customer list customers search result loading status.
-   */
-  getCustomerListCustomersSearchResultsLoading(): Observable<boolean> {
-    return this.store.pipe(
-      select(AsmSelectors.getCustomerListCustomersSearchResultsLoading)
-    );
-  }
-
   /**
    * Returns the customer list customers search result error status.
```
Comment (niehuayang) [inline feature-libs/asm/core/facade/asm.service.ts]: Hi @ercultimate , It looks like the method should also move to AsmCustomerListFacade? Am I right?

File: feature-libs/asm/core/connectors/asm.adapter.ts (hunk 1 above)
Comment (niehuayang) [reply feature-libs/asm/core/connectors/asm.adapter.ts]: > This is looking pretty good. Just the imports to fix and then documenting the API changes in the migration document.

the API changes have been added in the migration document.

File: feature-libs/asm/core/facade/asm.service.ts (hunk 2 above)
Comment (ercultimate) [reply feature-libs/asm/core/facade/asm.service.ts]: Yes, you are correct!  I think this method was added later and might have been left out of the ticket description.

**Review by ercultimate — CHANGES_REQUESTED** [review]

File: feature-libs/asm/core/facade/asm.service.ts (hunk 2 above)
Comment (niehuayang) [reply feature-libs/asm/core/facade/asm.service.ts]: Okay, I have changed it

**Review by ercultimate — DISMISSED** [review]

**Comment by niehuayang** [comment]
Hi @ercultimate , When I want to merge to develop,  the code conflicts appear, I have solved the conflicts, could you please approve it again?

**Review by kimhw0630 — APPROVED** [review]

**Pull request merged**